### ISSUE TITLE
Project pulse is flatline.

### ISSUE THREAD

**Issue by Thrameos** [jython/jython3#36]
I was looking for a solution to using mixed python and java, but required python 3 for my company project.  When I started work on the project last spring, this project looked like a long road so I decided to use jpype.  I figured that this project would catch up eventually.  But since then it appears this project has completely stalled.  

Is there a road map or a task list with things that contributors can do to get this project back to moving?  Or has it died completely? What level of push is required to get a beta for python3?

**Comment by stuaxo** [comment]
Can't talk for jython3, but when it comes to jythons python2 support, most of the time it made big leaps were when companies sponsor it (I think they basically brought on the main jython developer for a bit or contracted him), that's how it made big leaps from 2.2, to 2.7.

**Comment by fwierzbicki** [comment]
Jython3 is more stalled than dead. There where a few years with a few different companies that allowed some of us to put more time into it and so things moved much faster, but we're all volunteers lately, so things are slower. We're currently focusing on a next 2.7.x, followed by a plan to move mainline development to github (this jython3 repo is experimental, but contains some really great progress. It is however a long way from a real release). There is no concrete roadmap for Jython3, that would come after we move to github and make the decision to move main development from the 2.x series to the 3.x series.

**Comment by Thrameos** [comment]
Thank you for your reply.

A road map of tasks that could be accomplished would certainly help it people wanted to make contributions.  I was looking to see if I could pitch some time into help, but it doesn't look active enough for me to be able to push it forward much.  

And stalled is often very similar to dead.  The more effort focuses on 2.7 the less relevant that work in this branch unless the drift rate is pretty slow.  I am sure it can all be backported, but that means any effort is limited to the person who knows what was done to diverge the two.  CPython is moving onto version 3.8 while this effort was targeting 3.5.  At least this project should revise its target and put down a road map of tasks needed so that it is clear what "long way" means.  The current state means that any potential contributors will likely do what I did and find easier paths.

**Comment by xforwad** [comment]
Really Dead?

**Comment by xforwad** [comment]
i wanna it back

**Comment by xforwad** [comment]
how can i be a contributor?

**Comment by Thrameos** [comment]
Certainly.  I am unfortunately on a project sabbatical due to manpower issues at my job so I can’t contribute right now. Due to a change in the institutional firewall I can’t even sign in to access the git or manage issues while at work.  Of course looking at my contribution log I always have a big drop off from November to June, though this year is unusually bad.  But I am trying to arrange for time starting in May timeframe.  There were only three items remaining for the 0.7 release, then I need to start on getting everything to work on the current Java 9+ releases.

If you can get the development version test it out and contribute updating the documentation it would be a big help.

**Comment by jbfaden** [comment]
I'd be happy to help in any way that I can.  Jython is an essential part of software I provide, and I know at some point the scientists who use it will be looking for Jython 3.

**Comment by Thrameos** [comment]
Actually disregard my last message.  I thought this was on the JPype messages that I am a contributor but am way behind thus my apologies to people I am behind on.  As far as I know Jython 3 is dead.  I wanted to contribute by no one appeared to be in charge.

**Comment by qunaibit** [comment]
GraalPython (https://github.com/graalvm/graalpython) might be a viable alternative, and it is actively maintained.

**Comment by jeff5** [comment]
It's not dead, it's resting. 🦜

Let's not mistake a repository for a project: all energy the project has is directed at 2.7.2b. I'd love to work on a 3.x.

What you see here is what it says it is, a sandbox is which potentially useful work was done, but on a fork. A look at all the fixes to 2.7 since the fork should convince you that some cruel bugs have been fixed to which this shot at Jython 3 is surely still heir.

**Comment by Phoenyx33** [comment]
Now that python 2.7 is definitely EOL, is anyone going to start work on Jython 3 since the scientists are going to start clamoring for it?

**Comment by jeff5** [comment]
Yes.

**Comment by pbailly** [comment]
Hello,

I am using Jython to write Python code and execute it inside a Java/BigData environment.
With the deprecation of Python 2.X getting closer, I want to migrate to something compatible with Python 3.X code.

I saw that [Jython project](https://hg.python.org/jython/) is still active and even got the release 2.7.2b1.
But I cannot find any news related to Python 3.X since your last message.

Do you have plan to focus on Jython3?

Thank you for any feedback!

**Comment by jeff5** [comment]
Yes, if we can get 2.7.2 out (even a beta) we hope to have some time for Jython 3. The work in this repository is a fork that, unfortunately, has diverged from 2.7 and it's not immediately clear how to recover the value in both together.

**Comment by predatorian3** [comment]
So, now that Python 2.7 is EOL as of 2020-01-01, will this project make a more concentrated effort on Jython 3?

**Comment by jeff5** [comment]
Quietly working on it, when not visibly working on 2.7.2.

**Comment by MauroGonzalez96** [comment]
hi, is there a date for jython 3?? or you recommend me to use python 2.7 for an applications for now??

**Comment by carloreggiani** [comment]
Any news? I'm getting: 

`DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support`

**Comment by jeff5** [comment]
https://github.com/jython/jython/issues/24#issuecomment-683253400
